sdk/macro/src/lib.rs

```diff
@@ -445,3 +447,266 @@ pub fn derive_clone_zeroed(input: proc_macro::TokenStream) -> proc_macro::TokenS
     }
     .into()
 }
+
+/// Macro for accessing data from the `package.metadata` section of the Cargo manifest
+///
+/// # Arguments
+/// * `key` - A string slice of a dot-separated path to the TOML key of interest
+///
+/// # Example
+/// Given the following `Cargo.toml`:
+/// ```ignore
+/// [package]
+/// name = "MyApp"
+/// version = "0.1.0"
+///
+/// [package.metadata]
+/// copyright = "Copyright (c) 2024 ACME Inc."
+/// ```
+///
+/// And the following `lib.rs`:
+/// ```ignore
+/// use solana_sdk_macro::package_metadata;
+///
+/// pub fn main() {
+///     let copyright = package_metadata!("copyright");
+/// }
+/// ```
+///
+/// Invoking `cargo run` will produce:
+/// ```ignore
+/// Copyright (c) 2024 ACME Inc.
+/// ```
```

Comment:
Not sure if you want to reword this without the print in there now.

Reply:
Ah yeah, good call, fixed itup